### PR TITLE
add: sort search feature and theme context

### PR DESCRIPTION
**Pull request by ApurvaSawant11**
In this PR, I have added the below features:

- Dark and Light theme
- Sort by Video upload date
- Filter videos by category
- Search video

### CODE REVIEW

**Review by ravikumar1002 — APPROVED** [review]

File: src/components/searchBar/searchBar.css [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .search-container {
     position: relative;
-    width: 40%;
+    /* width: 40%; */
```
Comment (anik31) [inline src/components/searchBar/searchBar.css]: suggestion - you can remove commented code if not required

File: src/components/searchBar/searchBar.css [hunk 2]
```diff
@@ -11,4 +11,10 @@
...
     left: 1rem;
-}
\ No newline at end of file
+}
+
+/* @media screen and (max-width: 860px) {
+    .search-container{
+        width:100%
+    }
+} */
```
Comment (anik31) [inline src/components/searchBar/searchBar.css]: suggestion - remove these also

File: src/context/theme-context.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    () => setTheme(localStorage.getItem("DazzleTube Theme") ?? "night"),
+    []
+  );
+
+  const changeTheme = () => {
+    localStorage.setItem(
+      "DazzleTube Theme",
+      theme === "night" ? "light" : "night"
+    );
+    setTheme(localStorage.getItem("DazzleTube Theme"));
```
Comment (anik31) [inline src/context/theme-context.js:19]: appreciation - liked the implementation of dark mode

**Review by anik31 — APPROVED** [review]
Great work, UI is very nice. All the features are working perfectly.

**Comment by ApurvaSawant11** [comment]
> Great work, UI is very nice. All the features are working perfectly.

Thanks for reviewing. I have removed the commented code.

**Pull request merged**